akka-persistence/src/main/scala/akka/persistence/snapshot/local/LocalSnapshotStore.scala

```diff
@@ -148,10 +149,7 @@ private[persistence] class LocalSnapshotStore extends SnapshotStore with ActorLo
 
   private final class SnapshotFilenameFilter(persistenceId: String) extends FilenameFilter {
     def accept(dir: File, name: String): Boolean = {
-      name match {
-        case FilenamePattern(pid, snr, tms) ⇒ pid.equals(URLEncoder.encode(persistenceId))
-        case _                              ⇒ false
-      }
+      extractMetadata(name).exists(_._1.equals(persistenceId))
```

Comment:
Maybe even a special method for this, avoiding the extra work and allocating a Some for each file, compared to a single `indexOf` and a `startsWith`.

Also, url-encode the persistence id in the constructor of this inner class instead, then it will be very clear that is encoded, I think the responsibility anyway belong together with this logic which "knows" that it is needed.